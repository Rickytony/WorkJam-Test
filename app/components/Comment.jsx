import React from "react";
import PropTypes from "prop-types";
import ReactHtmlParser from "react-html-parser";

const Comment = ({ comment }) => {
  return (
    <div className="mt-2">
      <p className="mb-0">{comment.by}:</p>
      <div>{ReactHtmlParser(comment.text)}</div>
    </div>
  );
};

Comment.propTypes = {
  comment: PropTypes.shape({
    by: PropTypes.string.isRequired,
    text: PropTypes.string.isRequired
  })
};

export { Comment };
