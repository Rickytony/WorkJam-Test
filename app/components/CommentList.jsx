import React from "react";
import PropTypes from "prop-types";
import { Comment } from "./Comment.jsx";

const CommentList = ({ comments, handleClick, showComments }) => {
  if (comments.length < 1)
    return (
      <div className="col-6 mt-2 mb-3">
        <h6>No comments yet.</h6>
      </div>
    );
  return (
    <div className="col-sm-12 col-md-6 mt-2 mb-4">
      <button
        type="button"
        onClick={e => handleClick(e)}
        className="btn btn-secondary mt-1"
      >
        Comments ({comments.length})
      </button>

      {showComments &&
        comments &&
        comments.map(comment => <Comment comment={comment} key={comment.id} />)}
    </div>
  );
};

CommentList.propTypes = {
  comments: PropTypes.array.isRequired,
  showComments: PropTypes.bool.isRequired,
  handleClick: PropTypes.func.isRequired
};

export { CommentList };
