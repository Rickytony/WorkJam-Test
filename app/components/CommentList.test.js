import React from "react";
import { shallow } from "enzyme";
import { CommentList } from "./CommentList";
import { Comment } from "./Comment";

describe("CommentList", () => {
  const handleClickMock = jest.fn();
  const showComments = false;
  const comments = [
    {
      by: "commenter",
      text: "comment about the article",
      id: 1
    },
    {
      by: "angry commenter",
      text: "angry comment about the article",
      id: 2
    },
    {
      by: "sarcastic commenter",
      text: "sarcastic comment about the article",
      id: 3
    }
  ];

  it("renders correctly when there are no comments", () => {
    const wrapper = shallow(
      <CommentList
        handleClick={handleClickMock}
        showComments={false}
        comments={[]}
      />
    );
    expect(wrapper.find("h6").text()).toEqual("No comments yet.");
  });

  it("renders correctly when there are no comments", () => {
    const wrapper = shallow(
      <CommentList
        handleClick={handleClickMock}
        showComments={false}
        comments={comments}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("calls handleClick when button is pressed", () => {
    const wrapper = shallow(
      <CommentList
        handleClick={handleClickMock}
        showComments={false}
        comments={comments}
      />
    );
    wrapper.find("button").simulate("click");
    expect(handleClickMock).toHaveBeenCalled();
  });

  it("renders correctly when there are no comments", () => {
    const wrapper = shallow(
      <CommentList
        handleClick={handleClickMock}
        showComments={true}
        comments={comments}
      />
    );
    expect(wrapper.find(Comment)).toHaveLength(3);
  });
});
