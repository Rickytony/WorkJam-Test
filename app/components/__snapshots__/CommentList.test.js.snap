// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentList renders correctly when there are no comments 1`] = `ShallowWrapper {}`;
